src/app/providers.tsx

```diff
@@ -0,0 +1,47 @@
+'use client'
+
+import { ReactQueryDevtools } from '@tanstack/react-query-devtools'
+import {
+    isServer,
+    QueryClient,
+    QueryClientProvider,
+} from '@tanstack/react-query'
+
+export function Providers({
+    children,
+}: Readonly<{ children: React.ReactNode }>) {
+    function makeQueryClient() {
+        return new QueryClient({
+            defaultOptions: {
+                queries: {
+                    refetchOnWindowFocus: false,
+                    retry: false,
+                    // With SSR, we usually want to set some default staleTime
+                    // above 0 to avoid refetching immediately on the client
+                    staleTime: 60 * 1000, // 1 min
+                },
+            },
+        })
+    }
+    let browserQueryClient: QueryClient | undefined = undefined
+
+    function getQueryClient() {
+        if (isServer) {
+            return makeQueryClient()
+        } else {
+            if (!browserQueryClient) browserQueryClient = makeQueryClient()
+            return browserQueryClient
+        }
+    }
```

Comment:
이 코드는 중첩된 if문 때문에 가독성이 좋지 않은것 같아요.
else 를 제거하고 단일 If문으로 처리하면 더 깔끔할것 같습니다. 대략 아래와 같은 흐름으로요.
```suggestion
function getQueryClient() {
    if (isServer) {
        return makeQueryClient();
    }

    if (!browserQueryClient) {
        browserQueryClient = makeQueryClient();
    }
    
    return browserQueryClient;
}
```